### PR TITLE
issue-207- help command message fix

### PR DESCRIPTION
**Pull request by DhanashriPalodkar**
**What type of PR is this?**
kind code-refactoring
Refactoring command help messages

**What does this PR do / why we need it**:
fixes --help command message around init.go, add.go, and bootstrap.gp

**Which issue(s) this PR fixes**:
issue-207,208 and 209

Fixes #?
https://issues.redhat.com/browse/GITOPS-207
https://issues.redhat.com/browse/GITOPS-208
https://issues.redhat.com/browse/GITOPS-209

**How to test changes / Special notes to the reviewer**:
use the following command

odo pipelines bootstrap --help
odo pipelines init --help
odo pipelines add --help

### CODE REVIEW

File: pkg/odo/cli/pipelines/init.go [hunk 1]
```diff
@@ -93,20 +93,22 @@ func NewCmdInit(name, fullName string) *cobra.Command {
...
-	cmd.Flags().StringVar(&o.StatusTrackerAccessToken, "status-tracker-access-token", "", "used to authenticate requests to push commit-statuses to your Git hosting service")
+	cmd.Flags().StringVar(&o.InternalRegistryHostname, "internal-registry-hostname", "image-registry.openshift-image-registry.svc:5000", "Internal image registry hostname")
+	cmd.Flags().StringVar(&o.ImageRepo, "image-repo", "", "Image repository in this form <registry>/<username>/<repository> or <project>/<app> for  for internal registry to push built images at sigle repository")
+	cmd.Flags().StringVar(&o.SealedSecretsService.Namespace, "sealed-secrets-ns", "sealed-secrets", "Namespace in which the Sealed Secrets operator is installed, automatically generated secrets are encrypted with this operator")
+	cmd.Flags().StringVar(&o.SealedSecretsService.Name, "sealed-secrets-svc", "sealedsecretcontroller-sealed-secrets", "Name of the Sealed Secrets Services that encrypts secrets")
+	cmd.Flags().StringVar(&o.StatusTrackerAccessToken, "status-tracker-access-token", "", "To authenticate requests to push commit-statuses to your Git hosting service")
```
Comment (bigkevmcd) [inline pkg/odo/cli/pipelines/init.go]: How about `Used to authenticate requests...` ?

File: pkg/odo/cli/pipelines/init.go [hunk 2]
```diff
@@ -93,20 +93,22 @@ func NewCmdInit(name, fullName string) *cobra.Command {
...
+	initCmd.MarkFlagRequired("gitops-repo-url")
 	return initCmd
 }
 
 func addInitCommands(cmd *cobra.Command, o *pipelines.InitOptions) {
 	cmd.Flags().StringVar(&o.GitOpsRepoURL, "gitops-repo-url", "", "Provide the URL for your GitOps repository e.g. https://github.com/organisation/repository.git")
 	cmd.Flags().StringVar(&o.GitOpsWebhookSecret, "gitops-webhook-secret", "", "Provide a secret that we can use to authenticate incoming hooks from your Git hosting service for the GitOps repository. (if not provided, it will be auto-generated)")
-	cmd.Flags().StringVar(&o.OutputPath, "output", ".", "Folder path to add GitOps resources")
-	cmd.Flags().StringVarP(&o.Prefix, "prefix", "p", "", "Add a prefix to the environment names")
+	cmd.Flags().StringVar(&o.OutputPath, "output", ".", "Folder path to add GitOps resources to save output")
```
Comment (bigkevmcd) [inline pkg/odo/cli/pipelines/init.go]: How about `Path to write GitOps resources`

**Review by bigkevmcd — CHANGES_REQUESTED** [review]

File: pkg/odo/cli/pipelines/init.go (hunk 1 above)
Comment (DhanashriPalodkar) [reply pkg/odo/cli/pipelines/init.go]: sounds right

File: pkg/odo/cli/pipelines/service/add.go [hunk 3]
```diff
@@ -67,17 +67,17 @@ func newCmdAdd(name, fullName string) *cobra.Command {
...
-	cmd.Flags().StringVar(&o.SealedSecretsService.Name, "sealed-secrets-svc", "sealedsecretcontroller-sealed-secrets", "name of the Sealed Secrets services that encrypts secrets")
+	cmd.Flags().StringVar(&o.GitRepoURL, "git-repo-url", "", "GitOps repository e.g. https://github.com/organisation/repository")
+	cmd.Flags().StringVar(&o.WebhookSecret, "webhook-secret", "", "Source Git repository webhook secret (if not provided, it will be auto-generated)")
+	cmd.Flags().StringVar(&o.AppName, "app-name", "", "Name of the application where the service will be added")
+	cmd.Flags().StringVar(&o.ServiceName, "service-name", "", "Name of the service to be added")
+	cmd.Flags().StringVar(&o.EnvName, "env-name", "", "Name of the environment where the service will be added")
+	cmd.Flags().StringVar(&o.ImageRepo, "image-repo", "", "Image repository in this form <registry>/<username>/<repository> or <project>/<app> for internal registry to push built images at sigle repository")
```
Comment (bigkevmcd) [inline pkg/odo/cli/pipelines/service/add.go]: s/sigle/single/

File: pkg/odo/cli/pipelines/service/add.go [hunk 4]
```diff
@@ -67,17 +67,17 @@ func newCmdAdd(name, fullName string) *cobra.Command {
...
+	cmd.Flags().StringVar(&o.GitRepoURL, "git-repo-url", "", "GitOps repository e.g. https://github.com/organisation/repository")
+	cmd.Flags().StringVar(&o.WebhookSecret, "webhook-secret", "", "Source Git repository webhook secret (if not provided, it will be auto-generated)")
+	cmd.Flags().StringVar(&o.AppName, "app-name", "", "Name of the application where the service will be added")
+	cmd.Flags().StringVar(&o.ServiceName, "service-name", "", "Name of the service to be added")
+	cmd.Flags().StringVar(&o.EnvName, "env-name", "", "Name of the environment where the service will be added")
+	cmd.Flags().StringVar(&o.ImageRepo, "image-repo", "", "Image repository of the form <registry>/<username>/<repository> or <project>/<app> for internal registry to push built images to repository")
+	cmd.Flags().StringVar(&o.InternalRegistryHostname, "image-repo-internal-registry-hostname", "image-registry.openshift-image-registry.svc:5000", "Host-name for internal image registry Eg. docker-registry.default.svc.cluster.local:5000, an integrated container image registry to store images built by source code")
```
Comment (bigkevmcd) [inline pkg/odo/cli/pipelines/service/add.go]:
```suggestion
	cmd.Flags().StringVar(&o.InternalRegistryHostname, "image-repo-internal-registry-hostname", "image-registry.openshift-image-registry.svc:5000", "Host-name for internal image registry e.g. docker-registry.default.svc.cluster.local:5000, used if you are pushing your images to the internal image registry")
```

File: pkg/odo/cli/pipelines/service/add.go [hunk 5]
```diff
@@ -67,17 +67,17 @@ func newCmdAdd(name, fullName string) *cobra.Command {
...
-
-	cmd.Flags().StringVar(&o.SealedSecretsService.Namespace, "sealed-secrets-ns", "sealed-secrets", "namespace in which the Sealed Secrets operator is installed, automatically generated secrets are encrypted with this operator")
-	cmd.Flags().StringVar(&o.SealedSecretsService.Name, "sealed-secrets-svc", "sealedsecretcontroller-sealed-secrets", "name of the Sealed Secrets services that encrypts secrets")
+	cmd.Flags().StringVar(&o.GitRepoURL, "git-repo-url", "", "GitOps repository e.g. https://github.com/organisation/repository")
+	cmd.Flags().StringVar(&o.WebhookSecret, "webhook-secret", "", "Source Git repository webhook secret (if not provided, it will be auto-generated)")
+	cmd.Flags().StringVar(&o.AppName, "app-name", "", "Name of the application where the service will be added")
+	cmd.Flags().StringVar(&o.ServiceName, "service-name", "", "Name of the service to be added")
+	cmd.Flags().StringVar(&o.EnvName, "env-name", "", "Name of the environment where the service will be added")
+	cmd.Flags().StringVar(&o.ImageRepo, "image-repo", "", "Image repository of the form <registry>/<username>/<repository> or <project>/<app> for internal registry to push built images to repository")
```
Comment (bigkevmcd) [inline pkg/odo/cli/pipelines/service/add.go]:
```suggestion
	cmd.Flags().StringVar(&o.ImageRepo, "image-repo", "", "Image repository of the form <registry>/<username>/<repository> or <project>/<app> which is used to push newly built images")
```
This should also be in ImageRepo above.

File: pkg/odo/cli/pipelines/init.go [hunk 6]
```diff
@@ -93,21 +93,23 @@ func NewCmdInit(name, fullName string) *cobra.Command {
 	}
 
 	addInitCommands(initCmd, o.InitOptions)
+
+	initCmd.MarkFlagRequired("gitops-repo-url")
```
Comment (bigkevmcd) [inline pkg/odo/cli/pipelines/init.go:97]: This is already marked as required https://github.com/rhd-gitops-example/odo/pull/157/files#diff-b14a2c841122a2c26a96c625e4f1f82eR114

File: pkg/odo/cli/pipelines/init.go [hunk 7]
```diff
@@ -93,21 +93,23 @@ func NewCmdInit(name, fullName string) *cobra.Command {
...
+	cmd.Flags().StringVar(&o.ImageRepo, "image-repo", "", "Image repository of the form '<registry>/<username>/<repository>' or '<project>/<app>' for internal registry to push built images to the repository")
+	cmd.Flags().StringVar(&o.SealedSecretsService.Namespace, "sealed-secrets-ns", "sealed-secrets", "Namespace in which the Sealed Secrets operator is installed, automatically generated secrets are encrypted with this operator")
+	cmd.Flags().StringVar(&o.SealedSecretsService.Name, "sealed-secrets-svc", "sealedsecretcontroller-sealed-secrets", "Name of the Sealed Secrets Services that encrypts secrets")
+	cmd.Flags().StringVar(&o.StatusTrackerAccessToken, "status-tracker-access-token", "", "Used to authenticategi requests to push commit-statuses to your Git hosting service")
+	cmd.Flags().BoolVar(&o.Overwrite, "overwrite", false, "Overwrites previously existing GitOps configuration(if any)")
```
Comment (bigkevmcd) [inline pkg/odo/cli/pipelines/init.go]:
```suggestion
	cmd.Flags().BoolVar(&o.Overwrite, "overwrite", false, "Overwrites previously existing GitOps configuration (if any)")
```

**Review by bigkevmcd — CHANGES_REQUESTED** [review]

**Review by bigkevmcd — APPROVED** [review]

**Pull request merged**